browser/js/products/products.js

```diff
@@ -33,6 +33,15 @@ app.config(function ($stateProvider) {
 // -- all products
 app.controller('ProductsCtrl', function ($scope, products) {
   $scope.products = products;
+
+  // Product category toggleability
+  $scope.showCategory = {};
+  Object.keys($scope.products).forEach( function(key) {
+    $scope.showCategory[key] = false;
+  });
```

Comment:
we can implement this using a simple angular filter (quite like search except in this case you're only searching for category)